src/hotspot/share/logging/logFileStreamOutput.cpp

```diff
@@ -146,6 +164,28 @@ int LogFileStreamOutput::write_internal(const LogDecorations& decorations, const
   return written;
 }
 
+int LogFileStreamOutput::write_internal(const LogDecorations& decorations, const char* msg) {
+  int msg_len = (int)strlen(msg);
+
+  // Do not do anything if foldmultilines has been specified
+  if (_fold_multilines) return write_internal_line(decorations, msg, msg_len);
+
+  char* dupstr = os::strdup_check_oom(msg, mtLogging);
+  char* tmp = dupstr;
+
+  while (*tmp != '\0') {
+    if (*tmp == '\n') {
+      *tmp = '\0';
+    }
+    ++tmp;
+  }
+  int written = write_internal_line(decorations, dupstr, msg_len);
+
+  os::free(dupstr);
+
+  return written;
```

Comment:
I don't think we need to dupstr this, do we? Can't we do the changes in the original string instead?

Reply:
We would lose the `const`ness of the original string. Intuitively I'm not sure either if a printing function should modify the string passed to it.

In any way, the effect of this PR in performance is really negligible, so "duplicating or not" would be a case of neatness more than optimisation. I'm not sure about what's neater, though, so I'll return the question to you

Reply:
We'll leave it for future work. It can definitely be done, but is not necessary yet.